lib/hunter/commands/send.rb

```diff
@@ -107,14 +107,22 @@ def prepare_payload
 
         hostname = Socket.gethostname
 
+        # Find MAC of the relevant interface
+        interface = `ip route get #{Config.target_host} | sed -n -e 's/^.*dev //p' | awk '{print $1}'`.chomp
+        if interface == "lo"
+          interface = `ip route get 8.8.8.8 | sed -n -e 's/^.*dev //p' | awk '{print $1}'`.chomp
```

Comment:
This is more of a educate-me question instead of a review comment; why are we relying on the Google DNS address to fetch the interface we want when the `hunt` node sends to itself? What ties the Google DNS to the "correct" interface?

Reply:
It's a side effect of how `ip route get` works, when you use it to query the device's own address it returns `lo` (which doesn't have a MAC address). The implementation for login nodes uses Google's DNS as an example to determine what the "default" interface it would use for connection is.

Reply:
That makes sense, but will it always be the interface that belongs to the cluster's private network? Perhaps this is a question for @ColonelPanicks.

Reply:
Seeing as it's a fallback if the route to the target_host is loopback I think it's somewhat reasonable. 

My concern is that it's using the `Config.target_host`... Does this resolve to the one provided on the CLI in the event of `hunter send` specifying `--server`?